### PR TITLE
Pricing Grid: Styling the header

### PR DESCRIPTION
**Pull request by niranjan-uma-shankar**
#### Proposed Changes

* This PR is one among several stacked PRs that implements the new pricing grid design. In https://github.com/Automattic/wp-calypso/pull/71372, we introduced a new signup flow /start/onboarding-2023-pricing-grid that is only available in dev and wpcalypso environments.
* The design for the pricing grid can be found in 2e991-pb. Please keep the Figma design open in a separate tab.
* The parent PR is https://github.com/Automattic/wp-calypso/pull/71407.
* This PR adds styling changes for the header section (above the CTA buttons) to look like in Figma.

Screenshot 2022-12-26 at 12 10 21 PM

#### What Not To Test

* Responsive screens. It will be implemented in a separate PR. Right now, test in a full width desktop browser viewport.
* Non-en. The first launch will be `en` only.

#### Known Issues

* The Premium plan does not have the styling to highlight it as Popular. This is fixed in a subsequent PR.
* Clicking the CTAs on the Free and Enterprise plan don't work and are fixed in a subsequent PR.

#### Testing Instructions

* Visit `/start/onboarding-2023-pricing-grid` and verify that the header section (screenshot shown in the description) matches the design in Figma. In particular, pay attention to the following:
    * Fonts - font sizes, font weights, font family, color and so on.
    * Margins and paddings
* Verify that the UI matches in both monthly and annual plan toggles.
* Verify that the price display looks good in different currencies - e.g. `IDR` and `JPY` . In currencies like `INR` and `MXN`, we offer a first year discount, so there's a strike-out text on the price. The currencies we support are in 2e9ad-pb.
* Verify that the production `onboarding` flow at `/start` is unaffected by UI changes.

### CODE REVIEW

File: client/my-sites/plan-features-2023-grid/header-price.jsx [hunk 1]
```diff
@@ -1,33 +1,22 @@
...
 import { Component } from 'react';
 import PlanPrice from 'calypso/my-sites/plan-price';
 
-const PLANS_LIST = getPlans();
-
-export class PlanFeatures2023GridHeader extends Component {
+export class PlanFeatures2023GridHeaderPrice extends Component {
 	render() {
-		return this.renderPlansHeaderNoTabs();
+		return this.renderPlansHeader();
```
Comment (southp) [inline client/my-sites/plan-features-2023-grid/header-price.jsx]: I know it's been this way, but it looks like we can just rename `renderPlansHeader()` as `render()` to remove this layer of function call :)

**Review by southp — APPROVED** [review]
Works as described. The code also provides a good foundation for further iteration for sure. I only left one minor feedback, but that's totally not a blocker :)

File: client/my-sites/plan-features-2023-grid/header-price.jsx (hunk 1 above)
Comment (niranjan-uma-shankar) [reply client/my-sites/plan-features-2023-grid/header-price.jsx]: Makes sense, cleaner code is always better. I've pushed a commit to address this.

File: client/my-sites/plan-features-2023-grid/style.scss [hunk 2]
```diff
@@ -98,19 +105,22 @@ $plan-features-sidebar-width: 272px;
 
 	.plan-features-2023-grid__header-title {
 		margin-bottom: 5px;
-		font-size: $font-title-small;
+		font-size: $font-title-large;
 		line-height: 0.7;
-		color: var(--studio-gray-80);
+		color: var(--studio-gray-100);
 		font-weight: 400;
```
Comment (claudiucelfilip) [inline client/my-sites/plan-features-2023-grid/style.scss:247]: Nitpick: this `font-weight` is unnecessary as it gets overwritten by the one in line 118.

**Review by claudiucelfilip — APPROVED** [review]
LGTM, I just left a non-blocking note and inline comment:
* it's probably not covered by this PR but the whole grid is left aligned on larger screens
Screenshot 2022-12-29 at 11 34 08

**Comment by claudiucelfilip** [comment]
Overlapping the design over the page I found a couple text alignment issues, most visible being the larger gab between the prices and timeframe line beneath:
Screen Shot on 2022-12-29 at 13-22-19

I think we can get closer to the design with the following adjustments:
```
.plan-features--signup .plan-features-2023-grid__header-tagline
+ line-height: 1.4;

.plan-features--signup .plan-features-2023-grid__pricing .plan-price
+ line-height: 1.8rem;
+ padding-bottom: 0;
```

Screen Shot on 2022-12-29 at 13-25-01

**Comment by niranjan-uma-shankar** [comment]
> it's probably not covered by this PR but the whole grid is left aligned on larger screens

Correct, this gets fixed in a downstream PR that adds responsiveness https://github.com/Automattic/wp-calypso/pull/71532, so when that is merged the alignment will look apt.

**Comment by niranjan-uma-shankar** [comment]
@claudiucelfilip Could you take a look at the downstream PR https://github.com/Automattic/wp-calypso/pull/71558? It includes some refactors and fixes and the gap between the price and timeframe is much reduced there.

**Pull request merged**